From 77fc884abc2d5d432d0ded829c09936b9dd50750 Mon Sep 17 00:00:00 2001
From: Liu Ying <Ying.Liu@freescale.com>
Date: Thu, 9 Apr 2015 12:43:36 +0800
Subject: [PATCH] MLK-10660-1 video: mxc ipuv3 fb: Fix _setup_disp_channel2
 bailout in ->set_par

We should uninitialize ipu channel only for !on_the_fly cases in the bail-out
path of _setup_disp_channel2() in ->fb_set_par(), because we should be able to
keep the display path un-touched or un-impacted in the on_the_fly cases even if
we return failure from _setup_display_channel2().

Signed-off-by: Liu Ying <Ying.Liu@freescale.com>
---
 drivers/video/mxc/mxc_ipuv3_fb.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/drivers/video/mxc/mxc_ipuv3_fb.c b/drivers/video/mxc/mxc_ipuv3_fb.c
index 2283382..23b04e6 100644
--- a/drivers/video/mxc/mxc_ipuv3_fb.c
+++ b/drivers/video/mxc/mxc_ipuv3_fb.c
@@ -1315,7 +1315,8 @@ static int mxcfb_set_par(struct fb_info *fbi)
 
 	retval = _setup_disp_channel2(fbi);
 	if (retval) {
-		ipu_uninit_channel(mxc_fbi->ipu, mxc_fbi->ipu_ch);
+		if (!on_the_fly)
+			ipu_uninit_channel(mxc_fbi->ipu, mxc_fbi->ipu_ch);
 		return retval;
 	}
 
-- 
1.8.0

